### ISSUE TITLE
option for log file

### ISSUE THREAD

**Issue by chumma12** [rg3/youtube-dl#6990]
Hi,
I know that the output from youtube-dl can be redirected to a log file.
I am looking for an log option of youtube-dl, so that I can configure it in youtube-dl.conf permanently and not bother with providing a log redirection for each download.
Infact, if the log file can also have options like -o parameter(then we can have log file with id in it's filename $id.log)

regards
Chumma

**Comment by yan12125** [comment]
What you need might be [writing a program embedding youtube-dl](https://github.com/rg3/youtube-dl#embedding-youtube-dl).

**Comment by chumma12** [comment]
All functionality can be achieved by the user writing his own program.  is that any kind of solution?
For a command line utility, writing to a log file is one of the basic functionalities.

**Comment by torrobinson** [comment]
Yeah I just had `youtube-dl`, after 2 years of use and it not having any issues, suddenly delete and re-download a bunch of stuff in a very weird way - and I can't seem to find a way to make it log so I can debug this in the future. Is there really no default logging option? Seems so trivial. It would be very helpful.

**Comment by marabu88** [comment]
I would also really like to see an option that allows to save a list of links that were not loaded due to some kind of error. In addition, recently I began to get a lock due to the frequent use of YouTube :( And this option has become even more relevant, since I don’t know which videos were not downloaded.

**Comment by sergeevabc** [comment]
Damn, make an option for logs at last! Sometimes ```youtube-dl``` is invoked in a peculiar way from 3rd party apps such as MPV and that “redirect output with 2>&1” advice is too tricky to implement there whereas a dedicated log switch would serve well.

**Comment by ryandesign** [comment]
Another good reason to add log-to-file functionality is that a user might want a separate logfile for each downloaded URL, but might want to invoke youtube-dl a single time and pass all of the URLs to it at once. For example, a user might want to download a set of videos, close the terminal, watch the videos later, and if there is a problem with any of the videos, the user could check the corresponding logfile that would be saved alongside the video (with the same name except .log extension).

Currently you just get a single stdout stream containing the information about all downloaded URLs with no obvious way to separate it by video file later.... Well, one workaround is to use something like `--exec echo {}`; then you know that any line starting with `[exec]` is the end of the file, and the following line is the name of the video file for the preceding log lines. So it's doable but not as straightforward as it could be if an option for logfiles were built in.

Even for the simpler use case of downloading a single URL, the fact that it can be accomplished with shell output redirection is not necessarily an argument against implementing it. After all, youtube-dl already implements a `--batch-file FILE` option which users could already accomplish using `xargs < FILE youtube-dl`.

**Comment by sorcerykid** [comment]
It's hard to believe after 7 years, such a basic (and much requested) functionality is still unavailable. This would probably take no more than 20 lines of Python code to implement in total. Check for config option, open log file for writing, duplicate STDOUT stream to log file, and lastly close log file. Sigh.
